app/models/charge_mode.rb

```diff
@@ -0,0 +1,12 @@
+class ChargeMode
+
+  def self.for_order_detail(order_detail)
+    if order_detail.price_policy
+      order_detail.price_policy.charge_for
+    else
+      current_policy = order_detail.product.current_price_policies.first
+      current_policy.try(:charge_for)
+    end
+  end
+
+end
```

Comment:
In that case, I have no problem making this kind of a service object for order details, but maybe I'd prefer an class with an instance method, not a class method. (Maybe even a SimpleDelegator).